📖 revise WEC registration based on feedback from #2997

rishi-jat opened the pull request:
## Summary  
This PR enhances the WEC registration documentation by incorporating all feedback provided in the review of PR #2997.

## What’s Improved  
- ✅ Mapped OCM terminology with references to official documentation  
- ✅ Corrected typo in the introduction (“what how to register” → “how to register”)  
- ✅ Improved explanation of verification commands  
- ✅ Reduced duplication across Kind, K3d, and OpenShift sections  
- ✅ Added clear cleanup instructions for both agent namespaces  
- ✅ Clarified labeling examples as illustrative only  
- ✅ Included detailed troubleshooting and management sections  
- ✅ Refined structure and overall readability for better comprehension

## Live Preview  
Preview → https://rishi-jat.github.io/kubestellar/doc-wec-registration/direct/wec-registration/
## Related  
Addresses feedback from PR #2997 by @MikeSpreitzer

rishi-jat commented:
Hey @MikeSpreitzer ! 👋
I’ve addressed all the feedback from #2997 and here’s a live preview of the updated WEC registration docs:
📄 Preview → https://rishi-jat.github.io/kubestellar/wec-preview/direct/wec-registration/
Let me know if anything else needs refinement!

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,325 @@
+Registering a WEC in an ITS is the same process as registering a managed cluster with an OCM hub cluster. KubeStellar uses Open Cluster Management (OCM) for cluster registration and management.
+
+**Terminology Mapping:**
+- **OCM Hub** → **KubeStellar ITS** (Inventory and Transport Space)
+- **OCM Managed Cluster** → **KubeStellar WEC** (Workload Execution Cluster)
+- **OCM Agent** → **OCM Agent** (same component, installed on WEC)
```
This did not render as you intended.

<img width="782" alt="image" src="https://github.com/user-attachments/assets/3b53f651-a814-48a9-8753-21e580b5464b" />

Using a proper section heading would probably fix it.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
````diff
@@ -1,5 +1,325 @@
+kubectl --context <its-context> get ns customization-properties
+```
+
+## Registration Process
````
This is the same information as is in https://docs.kubestellar.io/unreleased-development/direct/get-started/#create-and-register-two-workload-execution-clusters . Rather than maintain two copies, we should maintain one copy and refer to it from elsewhere as needed.

MikeSpreitzer commented:
@Rupam-It: Since you worked on #2997, perhaps you should look at this.

rishi-jat commented:
/retest-required

rishi-jat commented:
Hi @MikeSpreitzer,
I’ve fixed all the suggested changes and updated the PR description with the preview link:
👉 https://rishi-jat.github.io/kubestellar/doc-wec-registration/readme/
Let me know if anything else is needed!

MikeSpreitzer commented:
I see that the gross problems with the navigation tree have been resolved.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,307 @@
+Before registering a WEC, ensure you have:
+
+1. **A running Kubernetes cluster** that will serve as the WEC
+2. **Network connectivity** between the WEC and the ITS
```
I think that it is true and worthwhile to say that what is required is that HTTPS connections can be opened from the WEC to the ITS.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,307 @@
+1. **A running Kubernetes cluster** that will serve as the WEC
+2. **Network connectivity** between the WEC and the ITS
+3. **`kubectl` access** to both the WEC and ITS
+4. **`clusteradm` CLI tool** installed ([installation guide](https://open-cluster-management.io/docs/getting-started/installation/start-the-control-plane/))
```
There are several containers and (virtual or real) machines in sight, so it would be helpful to say _where_ this tool needs to be installed.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
````diff
@@ -1,5 +1,307 @@
+
+Execute the join command on the WEC to initiate the registration process:
+
+```shell
+# Basic join command
+clusteradm join --hub-token <token> --hub-apiserver <api-server-url> --cluster-name <your-wec-name> --context <wec-context>
+```
````
This does not hint that there might be additional flags required; that could lead an eager reader to make a mistake.

MikeSpreitzer replied:
The current revision still starts by telling the reader to do something without warning about `--force-internal-endpoint-lookup`.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
````diff
@@ -1,5 +1,307 @@
+```
+
+## Next Steps
+
````
Shouldn't there be a pointer to the general big picture (https://rishi-jat.github.io/kubestellar/doc-wec-registration/direct/user-guide-intro/#the-full-story) somewhere in here?

rishi-jat commented:
I’ve addressed all the feedback and updated the PR accordingly. Please have a look and let me know if anything else is needed.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,272 @@
+
+You should see a CSR with your WEC name and status `Pending`. This CSR is created by the OCM registration agent running on your WEC.
+
+**Wait for CSR to appear:**
```
Stop making pseudo-headers with bold text. Use real section header syntax. Perhaps something like the following.

```
#### Loop to Wait for Certificate Signing Request
```

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
````diff
@@ -1,5 +1,272 @@
+kubectl --context <its-context> get csr
+```
+
+You should see a CSR with your WEC name and status `Pending`. This CSR is created by the OCM registration agent running on your WEC.
````
No, "You should see ..." is not necessarily true. The first few `kubectl --context <its-context> get csr` might NOT show the CSR with your WEC name and status `Pending`. You need to explain to this to the reader and explicitly tell them to repeat this `get` until the needed CSR appears.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
````diff
@@ -1,5 +1,272 @@
+clusteradm join --hub-token <token> --hub-apiserver <api-server-url> --cluster-name <your-wec-name> --context <wec-context>
+```
+
+**Important flags:**
+- `--cluster-name`: Choose a unique name for your WEC
+- `--context`: Specify the kubectl context for your WEC
+- `--singleton`: Use this flag if the WEC is a single-node cluster
+- `--force-internal-endpoint-lookup`: Required for Kind clusters and other local development setups
````
This does not render as you intended. Make a practice of proof-reading your rendering.

<img width="817" height="87" alt="image" src="https://github.com/user-attachments/assets/a8b99e72-38cd-4511-b084-a75616e33c01" />

MikeSpreitzer replied:
This is still not rendering as you intend. Compare the website preview

<img width="760" height="118" alt="image" src="https://github.com/user-attachments/assets/110b1745-a326-4e41-ac13-4566a984e977" />

with the direct view at GitHub:

<img width="725" height="180" alt="image" src="https://github.com/user-attachments/assets/55236bc6-6441-4b86-bb8d-b16d23ae29ff" />

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,272 @@
+
+Registering a WEC with an ITS follows the same process as registering a managed cluster with an OCM hub. KubeStellar uses Open Cluster Management (OCM) for cluster registration and management.
+
+For the complete OCM registration process, refer to the [official Open Cluster Management documentation](https://open-cluster-management.io/docs/getting-started/installation/register-a-cluster/). The instructions below provide a complete registration process that is more focused on KubeStellar-specific considerations.
```
The problem here is that it leads off with "For the complete OCM registration process, refer to ...". That in itself implies that what follows is not a complete registration process. That is not the distinction that you want to make. What follows _is_ a complete registration process. Do not start by implying otherwise. Start by saying what you mean. Even more importantly, don't start by talking about what is NOT coming. Start by talking about what _is_ coming. Finish with a reference to the _comprehensive_ OCM documentation. "comprehensive documentation" means something different from "complete process".

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,272 @@
+
+### Step 2: Join the WEC to the ITS
+
+Execute the join command on the WEC to initiate the registration process:
```
This section still starts by telling the reader unconditionally to issue a particular command; unfortunately, that is not necessarily the right command. The reader is being told to do something without first being given a clue that she may also need the `--singleton` or `--force-internal-endpoint-lookup` flags.

MikeSpreitzer reviewed (commented):
I have finished another round of review.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,279 @@
+
+**Important:** Before executing the join command, determine if you need additional flags based on your environment. The basic command below may not work for all cluster types.
+
+Execute the join command on the WEC to initiate the registration process. The exact command depends on your environment:
```
s/:/./

The colon implies that what follows is an elaboration on the remark before the colon, but that is not what actually follows.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,279 @@
+
+Registering a WEC with an ITS follows the same process as registering a managed cluster with an OCM hub. KubeStellar uses Open Cluster Management (OCM) for cluster registration and management.
+
+The instructions below provide a comprehensive registration process focused on KubeStellar-specific considerations. For additional OCM registration details, refer to the [official Open Cluster Management documentation](https://open-cluster-management.io/docs/getting-started/installation/register-a-cluster/).
```
The word "comprehensive" here is not helping this sentence. I would simply remove it.

MikeSpreitzer commented on docs/content/direct/wec-registration.md:
```diff
@@ -1,5 +1,279 @@
+- `--singleton`: Use this flag if the WEC is a single-node cluster
+- `--force-internal-endpoint-lookup`: Required for Kind clusters and other clusters with internal-only API server endpoints
+
+**Example for Kind clusters:**
```
Shouldn't this be a real section heading too?

MikeSpreitzer reviewed (approved):
This is close enough to merge, with further improvements done in follow-on PR(s).

/lgtm
/approve
Thanks!
